File: lib/utils/identity.dart
/// Canonical normalization for identity fields, applied at every WRITE boundary
/// (not just in readers). Phone = bare 10-digit (strip non-digits, last 10);
/// email = trimmed + lowercased. Mirrors the web util src/lib/identity.ts and
/// the SQL normalization in find_user_by_phone.
String normPhone(String? p) {
  final d = (p ?? '').replaceAll(RegExp(r'\D'), '');
  return d.length > 10 ? d.substring(d.length - 10) : d;
}

String normEmail(String? e) => (e ?? '').trim().toLowerCase();

File: src/lib/identity.ts
/**
 * Canonical normalization for identity fields, applied at every WRITE boundary
 * (not just in readers). Phone = bare 10-digit (strip non-digits, last 10);
 * email = trimmed + lowercased. Mirrors the Flutter util lib/utils/identity.dart
 * and the SQL normalization in find_user_by_phone.
 */
export const normPhone = (p: string | null | undefined): string =>
  (p ?? '').replace(/\D/g, '').slice(-10);

export const normEmail = (e: string | null | undefined): string =>
  (e ?? '').trim().toLowerCase();
